### ISSUE TITLE
Awkwardness with draw interaction and doubleclick zoom interaction

### ISSUE THREAD

**Issue by tschaub** [openlayers/openlayers#3610]
If you try to use the draw interaction to draw a single feature, it seems sensible to deactivate or remove the interaction in a `drawend` event handler.  This is troublesome if the doubleclick zoom interaction is also added to map.  People (often) doubleclick to end drawing, the `drawend` event is dispatched in a pointerup handler, the draw interaction is disabled/removed, and the doubleclick zoom interaction handles the doubleclick when it is dispatched.

**Comment by tschaub** [comment]
Things are even more awkward if you have a draw interaction and are listening for `singleclick`.

**Comment by adube** [comment]
It happened to me too.  In ended up removing the interaction after a delay of 251ms, the time required for the double-click event to be detected by the interaction which allows it to prevent the event to be propagated further.  It's hackish, but it worked for me.  I'd be interested in hearing a solution for this.

**Comment by tschaub** [comment]
It would be nice if the interaction's handling of events would cancel the timeout used in generating the `singleclick` event (in addition to stopping propagation to other interactions).

**Comment by skeenp** [comment]
I'm no javascript expert, so open to comments. Using the information in this thread and others on the net I'm able to achieve this using the following code. I'm putting it here as reference for others.

```javascript
//In body of code to setup draw interaction
...
select.on('drawend', onFinishSelection);
...

//Setup drawend event handle function
function onFinishSelection(evt) {
   //Call to double click zoom control function to deactivate zoom event
   controlDoubleClickZoom(false);
  //Delay execution of activation of double click zoom function
   setTimeout(function(){controlDoubleClickZoom(true);},251); 
}

//Control active state of double click zoom interaction
function controlDoubleClickZoom(active){
	//Find double click interaction
	var interactions = map.getInteractions();
	for (var i = 0; i < interactions.getLength(); i++) {
		var interaction = interactions.item(i);                          
		if (interaction instanceof ol.interaction.DoubleClickZoom) {
			interaction.setActive(active);
		}
	}
}
```

**Comment by nikolas** [comment]
Does openlayers now have a configuration option to address this, or has anything changed since this ticket was opened?

**Comment by ahocevar** [comment]
This was acutally fixed with #7526. You have to configure the Draw interaction with `stopClick: true`.

**Comment by nikolas** [comment]
thank you!
